plot_app/templates/main.js

```diff
@@ -25,6 +26,68 @@ function navigate(fragment) {
 
 {% if is_plot_page %}
 
+var cur_err_ids = {{cur_err_ids}};
+
+function init_error_labels(){
+    // initialize error labels
+
+    for(var i=0; i < cur_err_ids.length; i++){
+        if(!$("#error-label").children().eq(cur_err_ids[i]-1).prop('selected')){
```

Comment:
This makes some assumption:
- ordering: it assumes the order matches the definition in `error_labels_table`. But `for _, err_label in error_labels_table.items():` does not guarantee that.
- the keys in `error_labels_table` are numbered from 1 to N. Which is correct now, but imagine we remove an item and add new ones. We cannot change or re-use indexes, because they're stored in the DB. This means the 1-to-N will be invalid.

A solution is to store the index in `<option>` as an additional attribute, and then search for that here.

Reply:
you're right, I deliberately made that assumption since I wanted to have the table strongly connected to the select. But I agree it's probably better to do it the way you described.